lib/oidcstrategy.js

```diff
@@ -143,6 +143,22 @@ function Strategy(options, verify) {
       This is fine if you are expecting multiple organizations to connect to your app.
       Otherwise you should validate the issuer.`);
   }
+
+  // validate other necessary option items provided, we validate them here and only once
+  var itemsToValidate = objectTransform({
+    source: options,
+    pick: ['clientID', 'callbackURL', 'responseType', 'responseMode']
+  });
+  var validatorConfiguration = {
+    clientID: Validator.isNonEmpty,
+    callbackURL: Validator.isNonEmpty,
+    // callbackURL: Validator.isURL  // We allow http for now
```

Comment:
Done. Now we have isHttpURL, isHttpsURL and isURL. For callbackURL, http is allowed so developer can use localhost for local dev; other URLs belongs to AAD so for those we only allow https.